Source/NSURL+Kakapo.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  NSURL+Kakapo.swift
+//  KakapoExample
+//
+//  Created by Joan Romano on 31/03/16.
+//  Copyright © 2016 devlucky. All rights reserved.
+//
+
+import Foundation
+
+typealias URLInfo = (params: [String : String], queryParams: [String : String])
+
+/// parseUrl: Checks and parses if a given `handleURL` representation matches a `requestURL`. Examples:
+///
+///
+///     `/users/:id` with `/users/1` produces `queryParams: ["id" : "1"]`
+///     `/users/:id/comments` with `/users/1/comments` produces `queryParams: ["id" : "1"]`
+///     `/users/:id/comments/:comment_id` with `/users/1/comments/2?page=2&author=hector` produces `queryParams: ["id": "1", "comment_id": "2"]` and `params: ["page": "2", "author": "hector"]`
+///
+/// - Parameter handlerURL: the URL with dynamic paths
+/// - Parameter requestURL: the actual URL
+/// - Returns: URLParams
+func parseUrl(handlerURL: String, requestURL: String) -> URLInfo? {
```

Comment:
cool @joanromano but as you said we may use NSURL which already supports all http://nshipster.com/nsurl/ otherwise we are reinventing the wheel and probably not unit testing enough ?